main.c

```diff
@@ -759,7 +825,7 @@ main (int   argc,
 
           Solvable *s = pool_id2solvable (pool, p);
 
-          map_setall (pool->considered);
+          map_init_clone (pool->considered, &excludes);
```

Comment:
You create big memory leak because every time this will happen, old copy will stay around.